src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Shopsys\ShopBundle\Model\PickUpPlace;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Shopsys\ShopBundle\Model\Country\CountryFacade;
+use Shopsys\ShopBundle\Model\Transport\Transport;
+use Symfony\Bridge\Monolog\Logger;
+
+class PickUpPlaceFacade
+{
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected $em;
+
+    /**
+     * @var \Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceLoader
+     */
+    protected $pickUpPlaceLoader;
+
+    /**
+     * @var \Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceRepository
+     */
+    protected $pickUpPlaceRepository;
+
+    /**
+     * @var \Shopsys\ShopBundle\Model\Country\CountryFacade
+     */
+    protected $countryFacade;
+
+    public function __construct(
+        PickUpPlaceRepository $pickUpPlaceRepository,
+        EntityManagerInterface $em,
+        PickUpPlaceLoader $pickUpPlaceLoader,
+        CountryFacade $countryFacade
+    ) {
+        $this->pickUpPlaceRepository = $pickUpPlaceRepository;
+        $this->em = $em;
+        $this->pickUpPlaceLoader = $pickUpPlaceLoader;
+        $this->countryFacade = $countryFacade;
+    }
+
+    /**
+     * @param int $id
+     * @return null|\Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlace
+     */
+    public function getById($id)
+    {
+        return $this->pickUpPlaceRepository->getById($id);
+    }
+
+    /**
+     * @param \Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceData $pickUpPlaceData
+     * @return \Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlace
+     */
+    public function create(PickUpPlaceData $pickUpPlaceData)
+    {
+        $pickUpPlace = new PickUpPlace($pickUpPlaceData);
+        $this->em->persist($pickUpPlace);
+        $this->em->flush($pickUpPlace);
+        return $pickUpPlace;
+    }
+
+    /**
+     * @param string $searchQuery
+     * @param string[] $countryCodes
+     * @param string $transportType
+     * @return mixed
+     */
+    public function findActiveBySearchQueryAndTransportType($searchQuery, $countryCoodes, $transportType)
+    {
+        return $this->pickUpPlaceRepository->findActiveBySearchQueryAndTransportType(
+            $searchQuery,
+            $countryCoodes,
```

Comment:
The result will be one pickup place from one country but in the process of searching for pickup place it will filter all pickup places where shop is allowed to ship.

Reply:
I see, the same functionality as our selectbox for countries.